File: src/main/java/com/github/krystianmuchla/home/note/NoteResponse.java
package com.github.krystianmuchla.home.note;

import java.time.Instant;
import java.util.UUID;

public record NoteResponse(UUID id, String title, String content, Instant creationTime, Instant modificationTime) {
    public NoteResponse(final Note note) {
        this(note.id(), note.title(), note.content(), note.creationTime(), note.modificationTime());
    }
}

File: src/main/kotlin/com/example/skyr/note/api/NoteResponse.kt
package com.example.skyr.note.api

import com.example.skyr.note.Note
import java.time.Instant
import java.util.*

data class NoteResponse(
    val id: UUID? = null,
    val title: String,
    val content: String,
    val creationDate: Instant,
    val modificationDate: Instant
)

fun noteResponse(note: Note) =
    NoteResponse(
        note.id,
        note.title,
        note.content,
        note.creationTime,
        note.modificationTime
    )

fun noteResponses(notes: List<Note>?) = notes?.let { it -> it.map { noteResponse(it) } } ?: emptyList()
